### PR TITLE
bug: 2315 Contribution Month and TEST FAIL in MASTER

### LINKED ISSUE #2315: 🐛 [Bug] Contribution Month and TEST FAIL  in MASTER
## 🐛 Bugreport
check out the current master and always run the frontend. 
an error should occur. 

at Object.<anonymous> (src/components/Contributions/ContributionForm.spec.js:93:46)
unknown

### check also this 
unknown

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
the last and current month is displayed cleanly again. 

### Issues
- fixes #2315 

image

### CODE REVIEW

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 1]
```diff
@@ -134,7 +134,7 @@ export default {
       // sets the date to the 1st of the previous month
       let date = new Date(this.maximalDate) // has to be a new object, because of 'setMonth' changes the objects date
       date = new Date(date.setMonth(date.getMonth() - 1))
-      return new Date(date.getFullYear(), date.getMonth(), 1)
+      return new Date(date.getFullYear(), date.getMonth() - 1)
```
Comment (Mogge) [inline frontend/src/components/Contributions/ContributionForm.vue:137]:
```
     minimalDate() {
       const date = new Date(this.maximalDate)
       return new Date(date.setMonth(date.getMonth() - 1, 1))
     },
```

**Review by Mogge — CHANGES_REQUESTED** [review]
The tests were good, they do what they should. Add a test case for 31.10.2022

**Comment by Mogge** [comment]
Add another test case for 28.2.2023

**Comment by ogerly** [comment]
the problem is the 31st of a month ... minus one month there is no 31st. here lies the problem

**Comment by Mogge** [comment]
My code suggestion above solves the problem, @ogerly

File: frontend/src/components/Contributions/ContributionForm.vue (hunk 1 above)
Comment (ogerly) [reply frontend/src/components/Contributions/ContributionForm.vue]: done

**Comment by ogerly** [comment]
i have more tests for: 
- date with the 31st day of the month
- date with the 28th day of the month
- date with 29.02.2024 leap year

**Review by Mogge — APPROVED** [review]

**Pull request merged**